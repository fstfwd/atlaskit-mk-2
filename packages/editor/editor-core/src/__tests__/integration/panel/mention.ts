import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';

import { insertMentionUsingClick } from '../message-renderer/_mention-helpers';

BrowserTestCase(
  'can insert mention into panel using click',
  { skip: ['ie', 'safari', 'edge'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitFor(editable);
    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await insertMentionUsingClick(browser, '0');
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
