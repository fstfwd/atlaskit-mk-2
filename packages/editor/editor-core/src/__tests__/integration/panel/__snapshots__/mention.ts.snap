// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mention.ts Panel: Can insert mention using click 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "accessLevel": "",
                "id": "0",
                "text": "@Carolyn",
                "userType": null,
              },
              "type": "mention",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
